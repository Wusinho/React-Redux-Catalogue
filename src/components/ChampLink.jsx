import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import '../style/Book.scss';

const ChampLink = ({ champs }) => {
  const getChamps = Object.entries(champs).map((item) => (item[1]));
  const getPicture = (name) => `url("http://ddragon.leagueoflegends.com/cdn/img/champion/loading/${name}_0.jpg")`;

  return getChamps.map((champ) => (
    <div
      key={champ.key}
      className="card m-2 rounded-3 link__card"
      style={{
        minWidth: '300px',
        backgroundImage: getPicture(champ.id),
        backgroundColor: 'black',
      }}
    >
      <Link to={`/champs/${champ.id}`} key={champ.key}>

        <div
          className="card-body"
        >
          <h5 className="card-title">{champ.title}</h5>
          <h2 className="card-title">{champ.name}</h2>
        </div>
      </Link>

    </div>
  ));
};

export default ChampLink;

ChampLink.defaultProps = {
  book: {
    id: '',
    name: '',
    title: '',
  },
};

ChampLink.propTypes = {
  book: PropTypes.objectOf(Object),
};
