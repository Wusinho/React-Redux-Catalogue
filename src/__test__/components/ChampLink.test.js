import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import ChampLink from '../../components/ChampLink';

describe('Champ', () => {
  test('renders when passed an object', () => {
    const champs = {
      Aatrox: {
        blur: 'Something',
        id: 'Aatrox',
        title: 'the Dark',
      },
      Ahri: {
        blur: 'Something',
        id: 'Aatrox',
        title: 'the Fox',
      },
    };
    const element = render(
      <Router>
        <ChampLink champs={champs} />
      </Router>,
    );
    expect(element).toMatchSnapshot();
  });
  test('nothing is render if undefined object', () => {
    const champs = {};
    const element = render(
      <ChampLink champs={champs} />,
    );
    expect(element.container).toMatchObject({});
  });
});
