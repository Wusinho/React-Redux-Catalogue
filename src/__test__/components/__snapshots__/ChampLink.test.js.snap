// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Champ renders when passed an object 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card m-2 rounded-3 link__card"
        style="min-width: 300px; background-image: url(http://ddragon.leagueoflegends.com/cdn/img/champion/loading/Aatrox_0.jpg); background-color: black;"
      >
        <a
          href="/champs/Aatrox"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              the Dark
            </h5>
            <h2
              class="card-title"
            />
          </div>
        </a>
      </div>
      <div
        class="card m-2 rounded-3 link__card"
        style="min-width: 300px; background-image: url(http://ddragon.leagueoflegends.com/cdn/img/champion/loading/Ahri_0.jpg); background-color: black;"
      >
        <a
          href="/champs/Ahri"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              the Fox
            </h5>
            <h2
              class="card-title"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card m-2 rounded-3 link__card"
      style="min-width: 300px; background-image: url(http://ddragon.leagueoflegends.com/cdn/img/champion/loading/Aatrox_0.jpg); background-color: black;"
    >
      <a
        href="/champs/Aatrox"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            the Dark
          </h5>
          <h2
            class="card-title"
          />
        </div>
      </a>
    </div>
    <div
      class="card m-2 rounded-3 link__card"
      style="min-width: 300px; background-image: url(http://ddragon.leagueoflegends.com/cdn/img/champion/loading/Ahri_0.jpg); background-color: black;"
    >
      <a
        href="/champs/Ahri"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            the Fox
          </h5>
          <h2
            class="card-title"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
